src/Symfony/Bundle/FrameworkBundle/Command/AssetsInstallCommand.php

```diff
@@ -137,7 +137,7 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $validAssetDirs = [];
         /** @var BundleInterface $bundle */
         foreach ($kernel->getBundles() as $bundle) {
-            if (!is_dir($originDir = $bundle->getPath().'/Resources/public')) {
+            if (!is_dir($originDir = $bundle->getPath().$bundle->getPublicPath())) {
```

Comment:
This can be a BC break in case someone implements the `BundleInterface` from scratch instead of extending the `Bundle` class.

Reply:
@jvasseur that was what I thinked at first, when I pointed this PR to Symfony 5.

Reply:
But you can check with `method_exists()` if the new method is present. If it isn’t, fall back to `/Resources/public` and trigger a deprecation warning.

Reply:
Are you sure deprecation is a good idea? Inside a loop, one would get N warnings (where N = number of bundles) for a single deprecation

Reply:
Most bundles use the `Bundle` class anyway, so it shouldn't be that many. We could technically collect all bundles that don't implement the method and issue one warning at the end, but I don't think that this is really necessary.

Anyway, a deprecation warning needs to happen one way or another because the user needs to be made aware of that fact that certain bundles need to be updated.

Reply:
Also done here